### DISCUSSION TITLE
Is it possible to register ValueConverter app-wide?

### DISCUSSION THREAD

**Discussion by fitdev** [AvaloniaUI/Avalonia#12427] [Q&A]
Is it possible to make it so that a given instance of my `IValueConverter` implementation be automatically used for certain "input type -> expected type" bindings automatically, i.e. without the need to do `Property="{Binding Foo, Converter={x:Static MyConverter.Instance}}"`? Can I somehow "register" these expected conversions so that they are applied app-wide?

**Comment by rabbitism** [comment]
Maybe you can try to implement a custom markup extension.

**Reply by fitdev** [reply]
@rabbitism Thank you for suggestion. But how would either a Type Converter or a Markup Extension help Avalonia framework automatically convert certain types during binding? With a Markup Extension I still have to write it in XAML.

**Reply by timunie** [reply]
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Markup/Avalonia.Markup.Xaml/Converters/ColorToBrushConverter.cs 

Don't know how exactly it is used in XAML, but this may give you a starting point to explore it in source and maybe you can register it somehow there. If it's worth it, tbh I doubt it is.
